scripts/tiling/stitching.py

```diff
@@ -0,0 +1,256 @@
+""" Script for creating a stitched ROI MIBItiff from a folder of FOVs
+"""
+
+import numpy as np
+import os
+from mibidata import mibi_image as mi
+from mibitracker.request_helpers import MibiRequests
+from collections import OrderedDict
+import time
+from mibidata import tiff
+import json
+
+MAX_TRIES = 10
+
+
+def combine_entity_by_name(roi_fov_paths, cols, rows, enforce_square):
+    min_col = np.min(cols)
+    min_row = np.min(rows)
+    cols=[v-min_col+1 for v in cols]
+    rows=[v-min_row+1 for v in rows]
+    w = np.max(cols)
+    h = np.max(rows)
+
+    panel = None
+    out_img = None
+    for fov_i,roi_fov_path in enumerate(roi_fov_paths):
+        print(fov_i+1,roi_fov_path)
+
+        fov_img = tiff.read(roi_fov_path)
+        panel = fov_img.channels
+        fov_img = fov_img.data
+
+        img_shape = list(np.shape(fov_img))
+        ch_count = np.min(img_shape)
+        shape_2d = img_shape[:-1]
+
+        if out_img is None:
+            out_img = np.zeros((h*shape_2d[0], w*shape_2d[1], ch_count), dtype=fov_img.dtype)
+            out_img_shape = list(np.shape(out_img))
+
+        for ch_i in range(ch_count):
```

Comment:
Why loop through all the channels instead of using something like `out_image[startY:endY, startX:endX, :] = fov_img[:, :, :]`?

Reply:
It's an artifact of our internal script where this functions for situations where the files are separate and needed to be loaded and added separately. Updating to your recommendation.